react-components/src/components/BundleProductOptions/bundleProductOptions.js

```diff
@@ -0,0 +1,300 @@
+            return {
+                option_id: item.option_id,
+                required: item.required,
+                title: item.title,
+                type: item.type,
+                quantity: ['checkbox', 'multi'].includes(item.type) ? 1 : item.options.find(o => o.is_default).quantity,
+                options: item.options
+                    .sort((a, b) => a.position - b.position)
+                    .map(option => {
+                        const {
+                            id,
+                            can_change_quantity,
+                            quantity,
+                            label,
+                            product: {
+                                price_range: {
+                                    maximum_price: {
+                                        final_price: { currency, value }
+                                    }
+                                }
+                            }
+                        } = option;
+
+                        return {
+                            id,
+                            can_change_quantity,
+                            quantity,
+                            label,
+                            currency,
+                            price: value
+                        };
+                    }),
+                customization: item.options
+                    .filter(o => o.is_default)
+                    .map(o => {
+                        const {
+                            id,
+                            quantity,
+                            product: {
+                                price_range: {
+                                    maximum_price: {
+                                        final_price: { value, currency }
+                                    }
+                                }
+                            }
+                        } = o;
+
+                        currencyCode = currency;
+
+                        return {
+                            id,
+                            quantity,
+                            price: value
+                        };
+                    })
+            };
+        });
+        setBundleState({ selections, currencyCode, quantity: 1 });
+    };
+
+    const handleSelectionChange = (option_id, quantity, customization) => {
+        const { selections } = bundleState;
+        const selectionIndex = selections.findIndex(s => s.option_id === option_id);
+        const selection = selections[selectionIndex];
+        selections.splice(selectionIndex, 1, { ...selection, quantity, customization });
+
+        setBundleState({ ...bundleState, selections: [...selections] });
+    };
+
+    const changeBundleQuantity = e => {
+        const { value } = e.target;
+        setBundleState({ ...bundleState, quantity: value });
+    };
+
+    const renderItemOptions = item => {
+        const { option_id, required, quantity, options, type, customization } = item;
```

Comment:
You can use an `useMemo` here to avoid the whole `switch` evaluation during re-renders.

Reply:
Can't use `useMemo` since this function is called from JSX and passes `item` as a param.

Reply:
maybe I'm wrong but I don't know how would I use `useMemo` here since it depends on `item` and that is passed as param from within JSX